### PR TITLE
Added feature_locations and changed the appropriate models

### PR DESCRIPTION
**Pull request by mikestephens**

### CODE REVIEW

File: lib/buweb/feature_location.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  validate :preferred_image_height_is_number
+
+  scope :custom_search, -> query {
+    begin
+      if query.present?
+        q = Regexp.new(query.to_s, Regexp::IGNORECASE)
+        any_of({title: q})
+      end
+    rescue RegexpError
+      Feature.none
```
Comment (adamcrown) [inline lib/buweb/feature_location.rb]: This should be `FeatureLocation.none` shouldn't it?

Comment (mikestephens) [inline lib/buweb/feature_location.rb]: D'oh good catch

File: lib/buweb/feature_location.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    if preferred_image_width.present? && preferred_image_width.is_a?(Numeric)
+      errors.add :base, 'Preferred image width must be numeric.'
+    end
+  end
+
+  def preferred_image_height_is_number
+    if preferred_image_height.present? && preferred_image_height.is_a?(Numeric)
+      errors.add :base, 'Preferred image height must be numeric.'
+    end
+  end
```
Comment (adamcrown) [inline lib/buweb/feature_location.rb]: Shouldn't these be `!...is_a?(Numeric)`. Also there is a `validates_numericality_of` in Rails that already handles these sort of validations.

Comment (mikestephens) [inline lib/buweb/feature_location.rb]: TIL

**Pull request merged**